workitem/workitem_repository.go

```diff
@@ -110,14 +111,54 @@ func (r *GormWorkItemRepository) Load(ctx context.Context, spaceID uuid.UUID, wi
 	return ConvertWorkItemStorageToModel(wiType, wiStorage)
 }
 
+// LookupIDByNamedSpaceAndNumber returns the work item's ID for the given owner name, space name and item number
+// returns NotFoundError, ConversionError or InternalError
+func (r *GormWorkItemRepository) LookupIDByNamedSpaceAndNumber(ctx context.Context, ownerName, spaceName string, wiNumber int) (*uuid.UUID, *uuid.UUID, error) {
+	defer goa.MeasureSince([]string{"goa", "db", "workitem", "LookupIDByNamedSpaceAndNumber"}, time.Now())
+	log.Debug(nil, map[string]interface{}{
+		"wi_number":  wiNumber,
+		"space_name": spaceName,
+		"owner_name": ownerName,
+	}, "Loading work item")
+	query := fmt.Sprintf("select wi.id, wi.space_id from %[1]s wi "+
```

Comment:
well.. what the explain says is you have 3 table scans and don't hit an index at all. But that could be due to the limited amount of data you have in your db.